src/t2iapi/context/service.proto

```diff
@@ -158,6 +158,62 @@ service ContextService {
   rpc EnsembleContextIndicateMembershipWithIdentification (BasicHandleRequest)
       returns (EnsembleContextIndicateMembershipWithIdentificationResponse);
 
+  /*
+  Get all ensemble-ids for a provided pm:EnsembleContextDescriptor and return them.
+
+  The ensemble-id is an identifier specified by the USER that uniquely refers to an SDC PARTICIPANT ENSEMBLE. The list
+  shall contain all ensemble-ids needed to cover each SDC PARTICIPANT ENSEMBLE membership can be indicated in by a
+  pm:EnsembleContextState with @ContextAssociation = 'Assoc' of the provided pm:EnsembleContextDescriptor.
+  The ensemble-ids are valid until the next call of this manipulation.
+   */
+  rpc GetEnsembleIds (BasicHandleRequest)
+      returns (GetEnsembleIdsResponse);
+
+  /*
+  Indicate membership in an SDC PARTICIPANT ENSEMBLE for a provided ensemble-id by creating or adapting all needed
+  pm:EnsembleContextStates such that their @ContextAssociation is "Assoc" and return a list of their @Handles.
```

Comment:
Please also mention the provided descriptorHandle.

Reply:
done